### PR TITLE
Rename the "Feedback" submenu "Form Responses"

### PR DESCRIPTION
**Pull request by donnchawp**
The Feedback->Feedback sidebar menu doesn't look great as it is just repeating the text of the main menu item.
This change makes it clearer what feedback is going into this page. It's the contact form of the website.

#### Testing instructions:

Apply the patch and note that the "Feedback" sub menu of the "Feedback" main menu item has changed to "Contact Form".

#### Does this pull request change what data or activity we track or use?

No. It's a simple text change.

### CODE REVIEW

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-jetpack-admin-menu.php
Comment (mmtr) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/class-jetpack-admin-menu.php:143]: Note that this will only change the unified menu name for Jetpack sites, which is at the moment useless because [Nav Unification is disabled for those sites](https://github.com/Automattic/wp-calypso/blob/d56adcceafc0293ef87cc6a8cca47acb7f0dab76/client/state/selectors/is-nav-unification-enabled.js#L27-L31).

Does this PR seeks to change the menu name also for other sites (Simple and Atomic) and/or when Nav Unification is disabled? In that case, I think you should change the label set during the custom post type registration: https://github.com/Automattic/jetpack/blob/8ffcd878f35e27fe23b5e4ab8cbdf85a8d1be7f7/projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php#L174

**Comment by kraftbj** [comment]
Related https://github.com/Automattic/jetpack/pull/9674

It would be great if we could lock down a single term that we could use across all ecosystems.

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-jetpack-admin-menu.php
Comment (donnchawp) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-jetpack-admin-menu.php:143]: You're right. I could have sworn that changed the sub-menu on my simple site. I'll update the patch.

**Comment by tjcafferkey** [comment]
@donnchawp do you have a status update on this? Currently it's in progress on our board but we're not sure if anyone is actively working on it still?

**Comment by donnchawp** [comment]
@mmtr can you take another look at this one? I changed the Feedback post type title to "Contact Form". The Feedback type itself is still "feedback".

**Comment by mmtr** [comment]
@donnchawp this is what I get when switching to this branch in Jetpack Beta:

Screen Shot 2021-04-07 at 12 10 40

Is that expected? When I read the PR summary it seems that we only want to rename the submenu, not the parent menu? Should we rename the "No feedback found" message too?

Also, I'm not sure whether the "Contact form" is less confusing than "Feedback". AFAIK, this screen displays the messages received through the contact form blocks inserted on posts which is way there isn't any "Add new" button. If we change the menu to "Contact form" users may get confused since they can expect to be able to manage contact forms rather than contact forms responses.

**Comment by jeherve** [comment]
> If we change the menu to "Contact form" users may get confused since they can expect to be able to manage contact forms rather than contact forms responses.

I would agree. In addition to that, Jetpack's Form feature has become a bit more than just a contact form builder in the past few years; you can now use it to build other types of forms:

image

Maybe we should avoid putting an emphasis on "Contact" now?

**Comment by donnchawp** [comment]
@mmtr no, that's not what I was expecting. I'll update the branch.

@jeherve what about making the sub-menu "Messages" instead? Something other than "Feedback" so we can avoid the Double Feedback->Feedback in the menu.

**Comment by jeherve** [comment]
> what about making the sub-menu "Messages" instead?

I think that's better. Or "Form responses"? I don't know what would best fit here. I'd be curious to have input from Design on this.

**Comment by folletto** [comment]
> I think that's better. Or "Form responses"? I don't know what would best fit here. I'd be curious to have input from Design on this.

Given the block is called "Form" I think a name like "Form responses" is quite a good fit. Clear, matches the name, and generic enough to adapt. It also matches "Responses" in Google Forms, so we would get an halo effect from there too.

**Comment by donnchawp** [comment]
I updated the PR. The main menu now appears as "Feedback" while the sub menu is "Form Responses". It definitely looks better.

File: projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php
Comment (jeherve) [inline projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php:171]: Since we only use it once in this function, maybe we can set the string directly in `add_menu_page`, without that new varialble?

File: projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php
Comment (jeherve) [inline projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php:259]: You'll need to add a doc comment here to appease the linter :)

File: projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php
Comment (jeherve) [inline projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php:263]:
```suggestion
			$slug
```

**Review by jeherve — CHANGES_REQUESTED** [review]

File: projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php
Comment (donnchawp) [reply projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php:171]: It's used in `register_post_type` too which is why I left that in.

File: projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php
Comment (donnchawp) [reply projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php:263]: Good catch. I'll get that fixed.

**Comment by donnchawp** [comment]
PR updated again with a doc comment and fixed the typo. Thanks @jeherve.

I also found a bug in the old Crowdsignal (ex-Polldaddy) plugin. It moves the Feedback menu item into the Crowdsignal top level menu. I need to fix that too separately.

File: projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php
Comment (mmtr) [inline projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php:215]: I get the following error when testing this branch on my local Jetpack site:

```
Fatal error: Uncaught Error: Call to undefined function add_menu_page()
in /usr/local/src/jetpack-monorepo/projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php on line 207
```

Seems the `add_menu_page` function is not available in the class `Grunion_Contact_Form_Plugin` constructor. I think this can be solved by moving this piece of code to `rename_feedback_menu`.

Comment (donnchawp) [reply projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php]: Thanks @mmtr for testing it. I couldn't replicate that error but I've moved that code into the rename function.

**Review by tjcafferkey — DISMISSED** [review]
Changes LGTM. The requested feedback has been implemented and it works as expected. Although the PR needs to be rebased before merging. Thanks!

File: projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php
Comment (BogdanUngureanu) [inline projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php:206]: Hey 👋 , I've tested your changes on my local environment and it seems that the `Export CSV` submenu link is broken. One way to fix this, I think, is to call the method directly here. Not sure though what are the implications of this. 

Also (and unrelated to your changes), I looks like that `Export CSV` page displays a blank page.

**Review by mmtr — CHANGES_REQUESTED** [review]
Works great now, thanks @donnchawp.

Site | Before | After
--- | --- | ---
Jetpack | Screen Shot 2021-05-07 at 11 03 09 | Screen Shot 2021-05-07 at 11 04 20
Simple | Screen Shot 2021-05-07 at 11 04 51 | Screen Shot 2021-05-07 at 11 06 24
Atomic | Screen Shot 2021-05-07 at 11 07 12 | Screen Shot 2021-05-07 at 11 48 23

However, note how when there are form responses (in the "Atomic" row above), we display a form for exporting them as CSV which is still using the "Feedback" term. I think we need to change that as well.

The "Export CSV" menu is also broken on both Jetpack and Atomic sites:

Screen Shot 2021-05-07 at 11 46 57

Screen Shot 2021-05-07 at 11 47 10

**Comment by donnchawp** [comment]
@mmtr Thanks for catching that. I fixed that in the latest change.

**Review by mmtr — DISMISSED** [review]
Awesome, the export menu works as expected now. I'd say we can also rename a few "feedback" occurrences used in the export UI, but I'll leave the final decision to you: 

```diff
diff --git a/projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php b/projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php
index 6f398e273..66d17caa8 100644
--- a/projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php
+++ b/projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php
@@ -467,7 +467,7 @@ class Grunion_Contact_Form_Plugin {
 
                add_submenu_page(
                        $slug,
-                       __( 'Export feedback as CSV', 'jetpack' ),
+                       __( 'Export responses as CSV', 'jetpack' ),
                        __( 'Export CSV', 'jetpack' ),
                        'export',
                        'feedback-export',
@@ -928,13 +928,13 @@ class Grunion_Contact_Form_Plugin {
                ?>
 
                <div id="feedback-export" style="display:none">
-                       <h2><?php _e( 'Export feedback as CSV', 'jetpack' ); ?></h2>
+                       <h2><?php _e( 'Export responses as CSV', 'jetpack' ); ?></h2>
                        <div class="clear"></div>
                        <form action="<?php echo admin_url( 'admin-post.php' ); ?>" method="post" class="form">
                                <?php wp_nonce_field( 'feedback_export', 'feedback_export_nonce' ); ?>
 
                                <input name="action" value="feedback_export" type="hidden">
-                               <label for="post"><?php _e( 'Select feedback to download', 'jetpack' ); ?></label>
+                               <label for="post"><?php _e( 'Select responses to download', 'jetpack' ); ?></label>
                                <select name="post">
                                        <option value="all"><?php esc_html_e( 'All posts', 'jetpack' ); ?></option>
                                        <?php echo $this->get_feedbacks_as_options(); ?>

```

**Comment by donnchawp** [comment]
There's always one more thing. I'll change those too.

**Review by mmtr — DISMISSED** [review]
Thanks for the latest improvements, @donnchawp.

**Review by jeherve — APPROVED** [review]

**Comment by donnchawp** [comment]
r225586-wpcom

**Comment by kwight** [comment]
@donnchawp I had to revert D59315-code on dotcom: p1620840435472900-slack-CB0B2G43X

**Comment by donnchawp** [comment]
OK. I'll try to fix the problem. That will likely need a new Jetpack PR, or if I can't I'll create a new PR reverting this one.

**Comment by mmtr** [comment]
@donnchawp turns out the issue was caused by the "Feedback" menu not being registered on API requests done by Calypso.  

The sidebar menu is rendered in Calypso via an API request to the `wpcom/v2/admin-menu` endpoint which doesn't pass the `is_admin` check performed before registering the "Feedback" menu.

I fixed that in https://github.com/Automattic/jetpack/pull/19886, and manually created a new diff (D61595-code) which contains changes from both this PR and https://github.com/Automattic/jetpack/pull/19886.

**Comment by donnchawp** [comment]
Thanks @mmtr. I missed the pings for this and the other patch so apologies for not reviewing it.

**Pull request merged**
